flocker/node/agents/blockdevice.py

```diff
@@ -410,6 +425,13 @@ class DestroyBlockDeviceDataset(PRecord):
     dataset_id = field(type=UUID, mandatory=True)
     blockdevice_id = field(type=unicode, mandatory=True)
 
+    @classmethod
+    def from_state_and_config(cls, discovered_dataset, desired_dataset):
```

Comment:
:question: Should there be an interface for `from_state_and_config` that is provided by these classes?